app/src/androidTest/java/com/sdpteam/connectout/event/EventActivityTest.java

```diff
@@ -153,4 +167,31 @@ public void deleteEventTest() {
         model.deleteEvent(NULL_USER, eventTitle1);
         assertNull(fJoin(model.getEvent(TEST_EVENT.getId())));
     }
+
+    /*
+    @Test
+    public void checkingRegistrationIsDisabledPassedDeadline() {
+        EventFirebaseDataSource dataSource = new EventFirebaseDataSource();
+        Calendar calendar = android.icu.util.Calendar.getInstance(TimeZone.getTimeZone("GMT+1:00"));
+        calendar.set(android.icu.util.Calendar.YEAR, 2022);
+        calendar.set(android.icu.util.Calendar.MONTH, 3 - 1); // Calendar.MONTH starts from 0
+        calendar.set(android.icu.util.Calendar.DAY_OF_MONTH, 18);
+        calendar.set(android.icu.util.Calendar.HOUR_OF_DAY, 4);
+        calendar.set(android.icu.util.Calendar.MINUTE, 20);
+        calendar.set(android.icu.util.Calendar.SECOND, 0);
+        calendar.set(Calendar.MILLISECOND, 0);
+        long unixTimestamp = calendar.getTimeInMillis();
+
+        dataSource.saveEvent(new Event(FAKE_EVENT_ID, "TEST TITLE", "TEST DESCRIPTION FOR TEST IN DESCRIPTION OF TEST",
+                null, "TEST ORGANIZER ID", new ArrayList<>(Arrays.asList("TEST ORGANIZER ID")),
+                Long.MAX_VALUE, new Event.EventRestrictions(4, 20, unixTimestamp)));
+        waitABit();
+
+        onView(withId(R.id.event_join_button)).perform(ViewActions.click());
+
+        waitABit();
+
+        onView(ViewMatchers.withId(R.id.event_chat_btn)).check(matches(isNotClickable())); // check that the registration didn't occur
+        FirebaseDatabase.getInstance().getReference().child(DATABASE_EVENT_PATH).child(FAKE_EVENT_ID).removeValue();
+    }*/
```

Comment:
We need to check with @Lukovitch for these tests, his tasks this sprint is about fixing them...